### PR TITLE
Release v1.2.0

### PR DESCRIPTION
**Pull request by arminru**
* add message/queue tracing to SDK API + samples
* add unit tests for SDK stub (DummyImpl)
* re-organize project/solution for building and testing in CI
* clarify readme + doc and improve samples

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -246,19 +249,153 @@ string incomingDynatraceStringTag = ...; // retrieve from incoming call metadata
...
+
+outgoingMessageTracer.Start();
+try
+{
+    Message message = new Message();
+    message.Headers[OneAgentSdkConstants.DYNATRACE_MESSAGE_PROPERTYNAME] = outgoingMessageTracer.GetDynatraceByteTag();
+
+    SendResult result = MyMessagingSystem.SendMessage(message);
+
+    outgoingMessageTracer.SetCorrelationId(result.CorrelationId);
```
Comment (AlramLechnerDynatrace) [inline readme.md]: correlationId needs to be generated by application and must be set on message before SendMessage(). i know: need to clarify that in the Java sample ...

File: samples/MessageTracerSamples.cs [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            outgoingMessageTracer.Start();
+            try
+            {
+                Message message = new Message();
+                // transport the Dynatrace tag along with the message:
+                message.Headers[OneAgentSdkConstants.DYNATRACE_MESSAGE_PROPERTYNAME] = outgoingMessageTracer.GetDynatraceByteTag();
+
+                SendResult result = MyMessagingSystem.SendMessage(message);
+
+                outgoingMessageTracer.SetCorrelationId(result.CorrelationId);     // optional
```
Comment (AlramLechnerDynatrace) [inline samples/MessageTracerSamples.cs]: see above

**Comment by AlramLechnerDynatrace** [comment]
lgtm

**Comment by kristianmandrup** [comment]
Any chance you will add ability to add custom metadata or setting `applicationId` to be used by DynaTrace collector?

I was super excited to see [this post on using request attributes](https://www.dynatrace.com/news/blog/capture-any-request-attributes-using-oneagent-sdk/) but then discovered it was not yet implemented for the .NET SDK. Any timeline? What else can we do?

We have a Switchr API servicing multiple client apps and we need to trace the full application/service tag through at least our API gateway. Using the DynaTrace url selector (from dashboard) was not sufficient, as multiple apps use the same route, so... now resorting to this SDK for more finegrained control. 

PS: Please make post on .NET SDK for general purpose application flow tracking . 

Cheers.

**Comment by SonjaChevre** [comment]
hi @kristianmandrup - we are just about to release support for messaging and queues. Request attributes is one of the next feature we will tackle after that. 

BTW, this is the page where we track features and requests: https://answers.dynatrace.com/spaces/483/dynatrace-product-ideas/idea/198106/planned-features-for-oneagent-sdk.html 

However we can already capture request attributes on .NET method arguments - any chance that could help for your use case? https://www.dynatrace.com/support/help/monitor/transactions-and-services/request-attributes/how-do-i-capture-request-attributes-based-on-dotnet-method-arguments/

**Comment by kristianmandrup** [comment]
Yes, I've looked into that as well. We might go with the web request attribute checking for now though and will wait for SDK support to do any "code intrusion". 

Not sure I completely understand the method arguments approach - blog post please 8>)
Perhaps compare the 3 different approaches, pros vs cons (limitations vs effort vs control)

See my comments here: https://github.com/Dynatrace/OneAgent-SDK-for-dotnet/issues/5

Cheers! Looking forward to 1.3 then

**Pull request merged**